### ISSUE TITLE
Ability to set `completedOnboarding` on User creation and edit via API 

### ISSUE THREAD

**Issue by Shpadoinkle** [calcom/cal.com#16605]
### Is your proposal related to a problem?
Currently we can create users via the API,  as well as set their schedules, assign them to Teams etc.  Basically set them up for use.

But on the first impersonation of this user, it _always_ sends you through the onboarding step

this can cause problems  as the last step makes you set availability, even if you have already done this via the API, which overwrites what you have previously setup

In our current setup, the availabilities are managed by an admin team.  So the user themselves will never log into the platform.    Making this onboarding step an unneeded step.

### Describe the solution you'd like
- ability to set the `completedOnboarding` value to true on user creation API. 
- support for this on api v1 at this point in time is preferable  

### Describe alternatives you've considered
there are no other options besides manual update of database. 

### Additional context

(Write your answer here.)

### Requirement/Document

(Share it here.)

---
##### House rules
- If this issue has a `🚨 needs approval` label, don't start coding yet. Wait until a core member approves feature request by removing this label, then you can start coding.
  - For clarity: Non-core member issues automatically get the `🚨 needs approval` label.
  - Your feature ideas are invaluable to us! However, they undergo review to ensure alignment with the product's direction.
  - Follow Best Practices lined out in our [Contributor Docs](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
